lib/velveteen/worker.rb

```diff
@@ -43,8 +43,29 @@ def logger
     end
 
     def publish(payload, options = {})
+      validate_published_message!(payload, options[:routing_key])
       options[:headers] = build_headers(options)
-      Config.exchange.publish(payload, options)
+      Config.exchange.publish(payload.to_json, options)
```

Comment:
I think this change will break existing workers, which are handling the `to_json` explicitly. It seems like a gap in Velveteen's test suite that there are no failures from this change.

I do think we want Velveteen to handle that, but it feels out of scope for this PR. Can you move forward on schema validation without making this change here?

Reply:
totally, good call